aliquotmaf/subcommands/vcf_to_aliquot/extractors/genotypes.py

```diff
@@ -85,6 +81,34 @@ def extract(cls, var_allele_idx, genotype, alleles):
             bcount = list(genotype["BCOUNT"])
             depths = [bcount[b_idx[i]] if i in b_idx else None for i in alleles]
 
+        # Handle CaVEMan which provides genotype and counts for all nucleotides in forward and reverse strands
+        elif set(
+            ["GT", "FAZ", "FCZ", "FGZ", "FTZ", "RAZ", "RCZ", "RGZ", "RTZ"]
+        ).issubset(genotype.keys()):
+            var_allele = alleles[var_allele_idx]
+            var_f_count_name = f"F{var_allele}Z"
+            var_r_count_name = f"R{var_allele}Z"
+            var_count = genotype[var_f_count_name] + genotype[var_r_count_name]
+            ref_allele = [al for al in alleles if al != var_allele][0]
+            ref_f_count_name = f"F{ref_allele}Z"
+            ref_r_count_name = f"R{ref_allele}Z"
+            ref_count = genotype[ref_f_count_name] + genotype[ref_r_count_name]
+
+            # set depths of alleles
+            depths = [ref_count, var_count]
+
+            # set DP to be sum of all observed base counts
+            new_gt["DP"] = sum(
+                [
+                    genotype[i]
+                    for i in ["FAZ", "FCZ", "FGZ", "FTZ", "RAZ", "RCZ", "RGZ", "RTZ"]
```

Comment:
This list should be a global variable at the top of the module since it is fairly important and unchanging:

```
from typing import Final

CAVEMAN_GENOTYPES: Final[List[str]] = ["FAZ", "FCZ", "FGZ", "FTZ", "RAZ", "RCZ", "RGZ", "RTZ"]
```

Reply:
ah good idea, will do